### PR TITLE
fix: use dd4hep::long64 instead of size_t for field indices

### PR DESCRIPTION
**Pull request by wdconinc**
### Briefly, what does this PR introduce?
The index of fields in the cell ID is a dd4hep::long64 (long long), not a size_t (unsigned long).

### What kind of change does this PR introduce?
- [x] Bug fix (issue #__)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No.

### Does this PR change default behavior?
No.

### CODE REVIEW

**Comment by wdconinc** [comment]
Ref. https://github.com/eic/EICrecon/pull/514#issuecomment-1448806251

File: src/algorithms/calorimetry/ImagingPixelReco.h [hunk 1]
```diff
@@ -36,7 +36,7 @@ class ImagingPixelReco {
     // length unit (from dd4hep geometry service)
     double m_lUnit; // {this, "lengthUnit", dd4hep::mm};
     // digitization parameters
-    unsigned int m_capADC; // {this, "capacityADC", 8096};
+    unsigned int m_capADC; // {this, "capacitysize_tADC", 8096};
```
Comment (veprbl) [inline src/algorithms/calorimetry/ImagingPixelReco.h:39]:
```suggestion
    unsigned int m_capADC; // {this, "capacityADC", 8096};
```

Comment (wdconinc) [reply src/algorithms/calorimetry/ImagingPixelReco.h]: The disadvantage of using the middle mouse button for both scrolling and pasting...

**Comment by wdconinc** [comment]
#749 and http://github.com/AIDASoft/DD4hep/pull/1225 address this.

**Pull request closed without merging**